force-app/main/freelanceCommunity/lwc/hot_myServiceAppointments/hot_myServiceAppointments.js

```diff
@@ -197,4 +217,36 @@ export default class Hot_myServiceAppointments extends LightningElement {
             }
         ];
     }
+    handleStatusChange(event) {
+        console.log('handleStatusChange', event.detail);
+        if (event.detail.interviewStatus == 'FINISHED') {
+            getServiceAppointment({
+                recordId: this.recordId
+            }).then((data) => {
+                console.log(data.Status);
+                if (data.Status == 'Completed') {
+                    this.isFlowFeedback = true;
+                    this.flowfeedback =
+                        'Det er ikke mulig å oppdatere statusen etter at oppdraget er satt til Dekket. Kontakt formidler for å gi ytterligere informasjon om oppdraget.';
+                    this.isflow = false;
+                    this.isCancelButtonHidden = true;
+                }
+                if (data.Status == 'Canceled') {
+                    this.isFlowFeedback = true;
+                    this.flowfeedback = 'Oppdraget er avlyst';
+                    this.isflow = false;
+                    this.isCancelButtonHidden = true;
+                    this.serviceAppointment.Status = 'Canceled';
+                }
+                if (data.HOT_CanceledByInterpreter__c) {
+                    this.isFlowFeedback = true;
+                    this.flowfeedback = 'Du er nå ikke lenger tildelt oppdraget';
+                    this.isflow = false;
+                    this.isCancelButtonHidden = true;
+                    this.serviceAppointment.Status = 'None';
```

Comment:
Disse feilmeldingene kan vi fjerne. Bare skjul hele flowen når den er ferdig.

Reply:
Her kan man refaktorere koden og forkorte en del. holder det ikke å sette this.serviceAppointment.Status = data.Status?, i tillegg til de to booleanene? Trenger vel ikke disse tre if-setningene?